lib/Core/addedByUser.ts

```diff
@@ -1,8 +1,24 @@
 import { BaseModel } from "../Models/Model";
-
+import i18next from "i18next";
 export const USER_ADDED_CATEGORY_ID = "__User-Added_Data__";
 
-export default function addedByUser(catalogMember: BaseModel): boolean {
+export default function addedByUser(
+  catalogMember: BaseModel,
+  options: {
+    depth: number;
+  } = {
+    depth: 0
+  }
+): boolean {
+  const depth = options.depth;
+  if (depth > 100) {
+    console.error(
```

Comment:
not sure how often this is going to happen so don't want to raise a terriaerror with raisetouser:true here